skops/io/_numpy.py

```diff
@@ -15,22 +16,29 @@ def ndarray_get_state(obj, dst):
         "__module__": get_module(type(obj)),
     }
 
-    try:
-        f_name = f"{uuid4()}.npy"
-        with open(Path(dst) / f_name, "wb") as f:
-            np.save(f, obj, allow_pickle=False)
+    if obj.dtype != object:
+        try:
+            digest = joblib.hash(obj)
+            f_name = f"{digest}.npy"
+            if not os.path.exists(Path(dst) / f_name):
+                with open(Path(dst) / f_name, "wb") as f:
+                    np.save(f, obj, allow_pickle=False)
             res["type"] = "numpy"
             res["file"] = f_name
-    except ValueError:
+        except ValueError as exc:
+            msg = (
+                f"numpy arrays of type {type(obj)} and dtype {obj.dtype} are not "
+                "supported yet"
+            )
+            raise TypeError(msg) from exc
```

Comment:
should this be our unsupported thingy error?

Reply:
Done